src/api.rs

```diff
@@ -119,42 +110,54 @@ impl SongList {
                     return Ok(NowOnAir {
                         song: song.clone(),
                         img_url,
-                        end_time
-                    })
+                    });
                 }
             }
         }
 
-        let artist_val = &parsed_json["results"][0]["songfile"]["artist"];
-        let title_val = &parsed_json["results"][0]["songfile"]["title"];
+        let artist_val = &parsed_json["data"][0]["artist"];
+        let title_val = &parsed_json["data"][0]["title"];
 
         if let Value::String(artist) = artist_val {
             if let Value::String(title) = title_val {
-                let song = self.songs.iter().find(|s| {
-                    s.artist == *artist && s.title == *title
-                });
+                let song = self
+                    .songs
+                    .iter()
+                    .find(|s| s.artist == *artist && s.title == *title);
 
                 return match song {
                     Some(song_some) => Ok(NowOnAir {
                         song: song_some.to_owned(),
-                        img_url: None,
-                        end_time
+                        img_url: song_some.image.clone(),
                     }),
                     None => Ok(NowOnAir {
                         song: Song {
                             id: 0,
                             title: title.to_string(),
                             artist: artist.to_string(),
                             position: None,
-                            url: "".to_string()
+                            url: "".to_string(),
+                            image: None,
+                            prev_position: None,
                         },
                         img_url: None,
-                        end_time
-                    })
-                }
+                    }),
+                };
             }
         }
 
         Err(ErrorKind::GenericError)
     }
-}
\ No newline at end of file
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn it_works() {
```

Comment:
Deze test was waarschijnlijk voor debuggen, maar aangezien er niks ge-assert wordt (en waarschijnlijk flaky is) kan deze wel weg. Verder ziet het er prima uit :+1:

Reply:
Goed punt!